pkg/proxy/nftables/proxier.go

```diff
@@ -1621,6 +1621,10 @@ func (proxier *Proxier) syncProxyRules() {
 	if err != nil {
 		proxier.logger.Error(err, "nftables sync failed")
 		metrics.NFTablesSyncFailuresTotal.Inc()
+
+		// staleChains is now incorrect since we didn't actually flush the
+		// chains in it. We can recompute it next time.
+		proxier.staleChains = make(map[string]time.Time)
```

Comment:
we used to do this thing to avoid relocating memory, no?

`proxier.staleChains = proxier.staleChains[:0]`

Reply:
hm... you can't do that with a map through...
ah, no, apparently as of golang 1.21 you can do `clear(proxier.staleChains)`

Reply:
to be honest, I didn''t realize it was a map when I commented, just saw the `make` and remembered the allocation problems ... glad it helped anyway